external/flume-assembly/pom.xml

```diff
@@ -0,0 +1,135 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--
+  ~ Licensed to the Apache Software Foundation (ASF) under one or more
+  ~ contributor license agreements.  See the NOTICE file distributed with
+  ~ this work for additional information regarding copyright ownership.
+  ~ The ASF licenses this file to You under the Apache License, Version 2.0
+  ~ (the "License"); you may not use this file except in compliance with
+  ~ the License.  You may obtain a copy of the License at
+  ~
+  ~    http://www.apache.org/licenses/LICENSE-2.0
+  ~
+  ~ Unless required by applicable law or agreed to in writing, software
+  ~ distributed under the License is distributed on an "AS IS" BASIS,
+  ~ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  ~ See the License for the specific language governing permissions and
+  ~ limitations under the License.
+  -->
+
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+  <modelVersion>4.0.0</modelVersion>
+  <parent>
+    <groupId>org.apache.spark</groupId>
+    <artifactId>spark-parent_2.10</artifactId>
+    <version>1.5.0-SNAPSHOT</version>
+    <relativePath>../../pom.xml</relativePath>
+  </parent>
+
+  <groupId>org.apache.spark</groupId>
+  <artifactId>spark-streaming-flume-assembly_2.10</artifactId>
```

Comment:
1. Good point, we can probably exclude scala. Why are the Hadoop libraries
included? Definitely not through spark as spark-streaming is marked as
provided dependency?

2. The whole point of making the assembly JAR is to make it easy to run
spark streaming + flume applications, especially in python where the users
will not be creating mvn/sbt projects to include the dependencies in an
uber jar. The most convenient for python users who want to use flume stream
is to add --jar <single-JAR-with-flume-stuff>.jar. Hence flume and its all
its dependencies need to be included.




On Mon, Jul 6, 2015 at 2:19 PM, Marcelo Vanzin <notifications@github.com>
wrote:

> In external/flume-assembly/pom.xml
> <https://github.com/apache/spark/pull/6830#discussion_r33983627>:
>
> > +  ~ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
> > +  ~ See the License for the specific language governing permissions and
> > +  ~ limitations under the License.
> > +  -->
> > +
> > +<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
> > +  <modelVersion>4.0.0</modelVersion>
> > +  <parent>
> > +    <groupId>org.apache.spark</groupId>
> > +    <artifactId>spark-parent_2.10</artifactId>
> > +    <version>1.5.0-SNAPSHOT</version>
> > +    <relativePath>../../pom.xml</relativePath>
> > +  </parent>
> > +
> > +  <groupId>org.apache.spark</groupId>
> > +  <artifactId>spark-streaming-flume-assembly_2.10</artifactId>
>
> Well, two things:
>
>    - spark-streaming-flume, the existing artifact, has transitive
>    dependencies on flume. So if you add it using the ivy support in
>    spark-submit, you'd get those.
>    - Even if you want to add this assembly, it currently packages way
>    more than just flume. It includes all of Scala and Hadoop libraries and a
>    bunch of other things, as I mentioned above.
>
> So any way you look at it, there is still something to be fixed here.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/apache/spark/pull/6830/files#r33983627>.
>